byway/src/main/webapp/routepage.js

```diff
@@ -292,5 +292,45 @@ function createRecButton(rec){
   return recBtn;
 }
 
-/* exported initMap, generateRoute, placesService, map */
-/* global google, findPlace, getTripKeyFromUrl, configureTripKeyForPath */
\ No newline at end of file
+
+/** Creates a URL link to Google Maps based on the start/end point and route
+ *  @returns {String} routeLink url containing query params for the user’s route
+ */
+function generateRouteLink(){
+    let routeLink = "https://www.google.com/maps/dir/?api=1&travelmode=driving" 
+    routeLink += "&origin=" + start.name.split(' ').join('%20') + "&destination=" + 
+      end.name.split(' ').join('%20') + "&waypoints=";
+    for(let place of route){
+      routeLink += place.name.split(' ').join('%20') + "|";
```

Comment:
What about other characters that need to be escaped besides space? There should be a function that does this encoding for you (`encodeURIComponent`). Also, you should never build a complexly formatted string by manually concatenating values together if there is a tool that can build it for you: 

```
new URLSearchParams({
    api: 1,
    travelmode: 'driving',
    destination: 'wherever the place',
    origin: 'from somewhere?',
    waypoints: ['a place', 'b somewhere ??? along the way? @%'].join('|')
}).toString()

// outputs: "api=1&travelmode=driving&destination=wherever+the+place&origin=from+somewhere%3F&waypoints=a+place%7Cb+somewhere+%3F%3F%3F+along+the+way%3F+%40%25"
```

Anytime you need to build search params for a URL, just use UrlSearchParams with an object arg to specify the parameters, so you don't have to worry about escaping them or messing up the encoding.

Reply:
Changed, I took out encodeURIComponent() since you said urlSearchParams() handles encoding.